core/modules/xrdsvc/ChannelStream.cc

```diff
@@ -61,30 +64,30 @@ void ChannelStream::append(StreamBuffer::Ptr const& streamBuffer, bool last) {
     if (_closed) {
         throw Bug("ChannelStream::append: Stream closed, append(...,last=true) already received");
     }
-    LOGS(_log, LOG_LVL_DEBUG, "last=" << last
-         << " " << util::prettyCharBuf(streamBuffer->data, streamBuffer->getSize(), 10));
+    LOGS(_log, LOG_LVL_DEBUG, "seq=" << to_string(_seq) << " ChannelStream::append last=" << last
+         << " " << util::prettyCharBuf(streamBuffer->data, streamBuffer->getSize(), 5));
     {
-        std::unique_lock<std::mutex> lock(_mutex);
-        LOGS(_log, LOG_LVL_DEBUG, "Trying to append message (flowing)");
+        unique_lock<mutex> lock(_mutex);
+        LOGS(_log, LOG_LVL_DEBUG, "seq=" << to_string(_seq) << " Trying to append message (flowing)");
 
         _msgs.push_back(streamBuffer);
         _closed = last; // if last is true, then we are closed.
-        _hasDataCondition.notify_one();
     }
+    _hasDataCondition.notify_one();
 }
 
 
 /// Pull out a data packet as a Buffer object (called by XrdSsi code)
 XrdSsiStream::Buffer* ChannelStream::GetBuff(XrdSsiErrInfo &eInfo, int &dlen, bool &last) {
-    std::unique_lock<std::mutex> lock(_mutex);
+    unique_lock<mutex> lock(_mutex);
     while(_msgs.empty() && !_closed) { // No msgs, but we aren't done
         // wait.
-        LOGS(_log, LOG_LVL_DEBUG, "Waiting, no data ready");
+        LOGS(_log, LOG_LVL_DEBUG, "seq=" << to_string(_seq) << " Waiting, no data ready");
         _hasDataCondition.wait(lock);
     }
-    if (_msgs.empty() && _closed) { // We are closed and no more
-        // msgs are available.
-        LOGS(_log, LOG_LVL_DEBUG, "Not waiting, but closed");
+    if (_msgs.empty() && _closed) {
+        // It's closed and no more msgs are available.
+        LOGS(_log, LOG_LVL_INFO, "seq=" << to_string(_seq) << " Not waiting, but closed");
```

Comment:
I've noticed a change from the `DEBUG` level to `INFO`. How frequently this message is supposed to be printed? Does it report any useful information that has to be seen at the `INFO` level?

Reply:
Currently, maybe once per job. This function is what xrootd calls when it wants data. This message is printed when there is no more data to send.